hydra-cluster/src/HydraNode.hs

```diff
@@ -304,6 +305,11 @@ withHydraNode' ::
   (Handle -> Handle -> ProcessHandle -> IO a) ->
   IO a
 withHydraNode' chainConfig workDir hydraNodeId hydraSKey hydraVKeys allNodeIds pparams mGivenStdOut action = do
+  case os of
+    -- XXX: this is to not bind to port 5000 on darwin systems,
+    -- as it might already be in use on recent macOS devices.
+    "darwin" -> hydraNodeId `shouldNotBe` 0
+    _ -> pure ()
```

Comment:
Could: use `when`

Could: assert the `port` used for `RunOptions` to be more direct. i.e. add `port = fromIntegral $ 5_000 + hydraNodeId` to the `where` clause and check here:

```suggestion
  -- NOTE: AirPlay on MacOS uses 5000 and we must avoid it.
  when (os == "darwin") $
    port `shouldNotBe` 5000
```